File: src/MemoryEvent.cpp
#include <QtCore/QString>

#include "MemoryEvent.h"

namespace CnotiMind
{

	/**!

		Creates a memory event with a \a key name and a \a value. The event time
		is the local time.

	*/
	MemoryEvent::MemoryEvent(const QString& key, const QString& value):
		Node( key, value ),
		_time( time() )
	{

	}

	MemoryEvent::MemoryEvent(const QString& key, const QString& value, time_t time):
		Node( key, value ),
		_time( time )
	{

	}

	time_t MemoryEvent::time() const
	{
		return _time;
	}

	const QString& MemoryEvent::key() const
	{
		return _key;
	}

	const QString& MemoryEvent::value() const
	{
		return _value;
	}

}

File: src_ios/CnotiMind/MemoryEvent.m
//
//  MemoryEvent.m
//  CnotiMind
//
//  Created by Gonçalo Rodrigues on 11/04/26.
//  Copyright 2011 Cnotinfor. All rights reserved.
//

#import "MemoryEvent.h"

/**!
 Creates a memory event with a \a key name and a \a value. The event time
 is the local time.
 */

@implementation MemoryEvent

@synthesize value = _value;
@synthesize event = _event;

@synthesize time = _time;


- (id) initWithEventAndValue:(NSString*)aEvent value:(id)aValue
{
    if (self == [super init]) {
        _event = [[NSString alloc] initWithString:aEvent];
        _value = [[NSString alloc] initWithString:aValue];

        _time = [[NSDate date] timeIntervalSince1970];
    }
    return self;
}


- (id) initWithEventAndValueAndTime:(NSString*)aEvent value:(NSString*)aValue time:(NSString*)aTime
{
    if (self == [self initWithEventAndValue: aEvent value:aValue]) {        
        _time = [aTime floatValue];
    }
    return self;
}


- (id) initWithPerception: (Perception*)aPerception
{
    if (self == [super init]) {
        _event = aPerception.name;
        _value = aPerception.value;
        
        _time = [[NSDate date] timeIntervalSince1970];
    }
    return self;
}

- (NSString*) toXML
{
    NSString* xml;
    DLog(@"event: %@", _event);
    DLog(@"value: %@", _value);
    DLog(@"time: %f", _time);
    xml = [NSString stringWithFormat:@"\t<MemoryEvent event=\"%@\" value=\"%@\" time=\"%f\" />\n", _event, _value, _time];
    
    return xml;
}

- (void) dealloc
{
    [_event release];
    [_value release];
    
    [super dealloc];
}

@end
